File: example/src/main/java/com/uploadcare/android/OrderDialogFragment.java
package com.uploadcare.android;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.uploadcare.android.example.R;
import com.uploadcare.android.library.urls.Order;

public class OrderDialogFragment extends DialogFragment {

    public interface OrderSelectedListener {
        void onOrderSelected(Order order);
    }

    private OrderSelectedListener listener;

    public static OrderDialogFragment newInstance() {
        return new OrderDialogFragment();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            if (getParentFragment() != null) {
                listener = (OrderSelectedListener) getParentFragment();
            } else {
                listener = (OrderSelectedListener) context;
            }
        } catch (ClassCastException e) {
            throw new ClassCastException("Parent must implement ReportReasonListener");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext()).setTitle(R.string.order_dialog_title)
                .setItems(R.array.orders, ((dialog, which) -> orderSelected(which)))
                .setNegativeButton(android.R.string.cancel, null)
                .create();
    }

    private void orderSelected(int position) {
        Order order;
        switch (position) {
            case 1:
                order = Order.UPLOAD_TIME_DESC;
                break;
            case 2:
                order = Order.SIZE_ASC;
                break;
            case 3:
                order = Order.SIZE_DESC;
                break;
            case 0:
            default:
                order = Order.UPLOAD_TIME_ASC;
                break;

        }
        listener.onOrderSelected(order);
    }

}

File: example/src/main/java/com/uploadcare/android/example/fragments/OrderDialogFragment.kt
package com.uploadcare.android.example.fragments

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.uploadcare.android.example.R
import com.uploadcare.android.library.urls.Order

class OrderDialogFragment : DialogFragment() {

    private lateinit var listener: OrderDialogListener

    override fun onAttach(context: Context) {
        super.onAttach(context)

        listener = try {
            if (parentFragment != null) {
                parentFragment as OrderDialogListener
            } else {
                context as OrderDialogListener
            }
        } catch (e: ClassCastException) {
            throw ClassCastException("Parent must implement OrderDialogListener")
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(context!!).apply {
            setItems(R.array.orders) { dialog, which -> orderSelected(which) }
            setNegativeButton(android.R.string.cancel, null)
        }.create()
    }

    private fun orderSelected(position: Int) {
        val order: Order
        when (position) {
            1 -> order = Order.UPLOAD_TIME_DESC
            2 -> order = Order.SIZE_ASC
            3 -> order = Order.SIZE_DESC
            0 -> order = Order.UPLOAD_TIME_ASC
            else -> order = Order.UPLOAD_TIME_ASC
        }
        listener.onOrderSelected(order)
    }

    companion object {

        fun newInstance(): OrderDialogFragment {
            return OrderDialogFragment()
        }
    }
}

interface OrderDialogListener {
    fun onOrderSelected(order: Order)
}